stustapay/core/service/order.py

```diff
@@ -1,33 +1,52 @@
 import logging
-from typing import Optional
+from collections import defaultdict
+from typing import Dict, Optional, Tuple
 
 import asyncpg
+from fastapi import HTTPException
+from starlette import status
 
-from stustapay.core.schema.order import LineItem, NewOrder, Order, OrderID
+from stustapay.core.schema.account import Account, get_source_account, get_target_account
+from stustapay.core.schema.order import CompletedOrder, LineItem, NewOrder, ORDER_TYPE_SALE, Order
 from stustapay.core.schema.terminal import Terminal
 from stustapay.core.schema.user import Privilege, User
-from .dbservice import DBService, requires_user_privileges, with_db_transaction
+from stustapay.core.service.dbservice import DBService, requires_user_privileges, with_db_transaction
 
 logger = logging.getLogger(__name__)
 
 
 class OrderService(DBService):
     @with_db_transaction
     @requires_user_privileges([Privilege.cashier])
-    async def create_order(
-        self, *, conn: asyncpg.Connection, current_user: User, current_terminal: Terminal, order: NewOrder
-    ) -> OrderID:
-        order_id: int = await conn.fetchval(
-            "insert into ordr (status, cashier_id, terminal_id) values ('pending', $1, $2) returning id",
+    async def execute_order(
```

Comment:
not sure we should immediately execute a new order - we should rather create it in pending state so we can show it as draft/overview. then, on a confirmation (or rather once the bändchen was scanned), it should be booked, and that's a separate api request.

Reply:
We can add another api request for that. But the customer should be present in the order/create request, to check the funds and age immediatly